Fix various Yielder leaks.

gianm opened the pull request:
- CombiningSequence leaked the input yielder from "toYielder" if it ran
  into an exception while accumulating the last value from the input
  yielder.
- MergeSequence leaked input yielders from "toYielder" if it ran into
  an exception while building the initial priority queue.
- ScanQueryRunnerFactory leaked the input yielder in its
  "priorityQueueSortAndLimit" strategy if it ran into an exception
  while scanning and sorting.
- YieldingSequenceBase.accumulate chomped IOExceptions thrown in
  "accumulate" during yielder closing.

The first bug (in CombiningSequence) led to periodic stalls in CalciteQueryTest, because it led to a merge buffer leak in testMaxSubqueryRows.

I'm not sure if any of these bugs have production impacts, but, they might.

gianm commented:
> Nice catch. Do you think we can add some unit tests for these fixes? It seems not hard at least for `CombiningSequence`, `MergeSequence`, and `YieldingSequenceBase`.

I agree it'd be good, I just didn't have time this afternoon 🙂

I can come back to this a bit later and add tests, though.

jihoonson commented:
Oh CI actually doesn't like the low test coverage 😅 

```
ERROR: Insufficient line coverage of 35% (19/53). Required 65%.
ERROR: Insufficient branch coverage of 50% (2/4). Required 65%.
```

gianm commented:
Added various tests, and fixed another bug that occurs when `yielder.get()` throws an exception while MergeSequence is creating the initial priority queue.
